### DISCUSSION TITLE
11월 25일 (목)

### DISCUSSION THREAD

**Discussion by InChil2** [InChil2/Detection_of_lesions#25] [회의록]
모델 성능을 높이기 위한 조건
- 데이터 셋이 충분히 크고 라벨링이 정확하게 되어있어야함
- Background Image(탐지할 물체가 없는 데이터)를 넣어서 False Positives를 줄이는 효과를 볼 수 있다 전체 학습 데이터 셋 0~10% 정도
- Pretrained weight 사용
- Epoch는 300부터 시작
- 이미지 사이즈 coco = 640 작은 크기의 물체가 많은 수록 높거나 원래의 해상도 사이즈를 사용 할 수록 좋다
- 기본 값으로 저장된 모델 하이퍼 파라미터들은 data 폴더 안에 hyp.scratch.yaml에서 찾을 수 있음
- --evolve를 인자 값으로 줘서 하이퍼파라미터 값을 추출 할 수 있다 시나리오 50번 이상을 돌려야지 나온다

원인 파악 = 데이터 전처리가 제대로 이루어지지 않음

mAP가 높게 나오는 원인 = Validation Data에 대한 예측률이라 LB score(Test mAP)와 격차가 있음

샘플 모델 생성 = mAP가 데이터가 적으면 에포크를 높여서 학습을 해야하고 데이터가 많으면 에포크가 적어도 학습이 잘 된다.
- mAP가 0.000단위로 나와서 평가가 불가능한 수준

균등 데이터 테스트 (50 / 500)= 데이터가 충분하지 않아서 특정 레이블(적은 데이터로도 잘 학습된 레이블)에 쏠림 현상 발생
- 이 이상으로 데이터를 늘려서 하는거는 전체 데이터 셋을 학습하는 거와 차이가 없음

Faster RCNN = 다른 모델에 비해서 특별한 점 없음
Yolov5(s/m/l/x) = 모델의 사이즈 별로 정확도와 학습 속도가 Trade off 관계

**Comment by pkwon35** [comment]
# Yolo5_X_epoch50의 Predict한 test 이미지 확인

**Reply by pkwon35** [reply]
## << 잘 분류된 모델 확인 >>

![image](https://user-images.githubusercontent.com/25214976/143397714-010ed1ba-ffa4-4136-8384-48f8c4107697.png)

![image](https://user-images.githubusercontent.com/25214976/143397655-fa8d9aa8-7a34-4db2-aa24-2c821241aaec.png)

![image](https://user-images.githubusercontent.com/25214976/143394225-32ff1d4a-436f-4386-91a2-c449420c5e45.png)

![image](https://user-images.githubusercontent.com/25214976/143395002-4f8c944e-9119-42c4-ba80-e289c97e8b13.png)

![image](https://user-images.githubusercontent.com/25214976/143395065-81affec5-61fc-4af5-af4a-fd0ffa59df7d.png)

![image](https://user-images.githubusercontent.com/25214976/143395273-2e87b352-757f-420c-8150-c960ba14692c.png)

![image](https://user-images.githubusercontent.com/25214976/143395395-3c0452fd-9839-4ff2-bd59-1c943ec6b710.png)

![image](https://user-images.githubusercontent.com/25214976/143395152-b7155054-d042-4fcf-8223-c5f90865ae14.png)

![image](https://user-images.githubusercontent.com/25214976/143396146-42459944-290b-4562-96fa-5fe0111d1fa6.png)

![image](https://user-images.githubusercontent.com/25214976/143396326-95d92a33-a080-4184-b689-5b05a3d8869c.png)

**Comment by pkwon35** [comment]
## << 잘 못 분류된 모델 확인 >>


==============================================================

잘 되었되었지만, 바운딩박스가 너무 크다
![image](https://user-images.githubusercontent.com/25214976/143395224-a6a6d401-8c5f-4d07-aad9-59c4149193b3.png)


![image](https://user-images.githubusercontent.com/25214976/143395678-37bb47b4-1eb3-450b-8e0b-cf8be765e116.png)


![image](https://user-images.githubusercontent.com/25214976/143396615-70dcea33-aeb2-4a99-baf4-28f1e24c1236.png)



=================== test_202095.png ======================
![image](https://user-images.githubusercontent.com/25214976/143396911-b095ac27-6ca2-46ff-b2b9-a11484369e73.png)  



=================== test_202090.png ======================
![image](https://user-images.githubusercontent.com/25214976/143397072-8f3d3617-e3ff-4d67-89da-e9276663f4e0.png)



=================== test_202087.png ======================
![image](https://user-images.githubusercontent.com/25214976/143397260-d23e929a-1da0-4a96-a052-60784ac5572e.png)



====test_202075====
![image](https://user-images.githubusercontent.com/25214976/143398035-0164f8c6-f259-4c21-a36a-e2ba32908920.png)


====test_202061====
![image](https://user-images.githubusercontent.com/25214976/143398366-77d617a9-5c52-4c4d-90c5-14cc7bc2f544.png)

=====test_202039====
![image](https://user-images.githubusercontent.com/25214976/143398783-49ebf002-d964-48ed-b6b5-c5f93aecc21f.png)



====test_20205====
![image](https://user-images.githubusercontent.com/25214976/143398467-87741749-2e90-4bb8-8aec-290bca794559.png)


===test_202019===
![image](https://user-images.githubusercontent.com/25214976/143399399-965c09e3-99b5-4137-80c6-df0b4ea8bbd0.png)


== test_200003 ==
![image](https://user-images.githubusercontent.com/25214976/143400019-1c42fec8-cc8e-44b0-82df-ad4deb0dfb09.png)

==test_200075==
![image](https://user-images.githubusercontent.com/25214976/143401968-d28efc16-6438-44ff-9fa9-6441d90a5d42.png)

**Comment by pkwon35** [comment]
## << 탐지 안된 이미지 >>
![image](https://user-images.githubusercontent.com/25214976/143395343-7ddc4b95-1a4f-402d-86a2-fc1369cf41ef.png)



=====test_202078=======
![image](https://user-images.githubusercontent.com/25214976/143397780-cae79068-0436-473f-9b51-806aab75ef48.png)



====test_202076====
![image](https://user-images.githubusercontent.com/25214976/143397902-d428483d-7d85-4008-a390-6a17b11b2775.png)


==test_202020==
![image](https://user-images.githubusercontent.com/25214976/143399258-5afd8072-5e3d-43a6-9d8e-bf5832aea47f.png)



==test_200000==
![image](https://user-images.githubusercontent.com/25214976/143399923-aae66002-2cab-415a-9c76-d41ab6b8edce.png)
